src/styles/main.scss

```diff
@@ -1,3 +1,88 @@
-body {
-  margin: 0;
+// Colors
+$gray: #eee;
+$pink: #ff69b4;
+$black: #000;
+
+// Sizes
+$cell-size: 100px;
+$border-size: 1px;
+$font-size: 30px;
+$padding-size: 10px;
+
+// Calculation variables
+$calendar-width: (7 * $cell-size) + (8 * $border-size) + (2 * $padding-size);
+$column-count: 7;
+
+// Mixins
+@mixin hover-styles {
+  cursor: pointer;
+  background-color: $pink;
+}
+
+// Modifier classes
+$start-days: (
+  "mon" 0,
+  "tue" 1 * ($cell-size + ($border-size * 2)) + 1px,
+  "wed" 2 * ($cell-size + $border-size * 2) + 2px,
+  "thu" 3 * ($cell-size + $border-size) + 6px,
+  "fri" 4 * ($cell-size + $border-size) + 8px,
+  "sat" 5 * ($cell-size + $border-size * 2) + 5px,
+  "sun" 6 * ($cell-size + $border-size * 2) + 5px
+);
+@each $day, $margin in $start-days {
+  .calendar--start-day--#{$day} {
+    .calendar__day:first-child {
+      margin-left: $margin;
+    }
+  }
+}
+
+// Modifier classes for month length
+@for $i from 28 through 31 {
+  .calendar--month-length--#{$i} {
+    .calendar__day:nth-child(n+#{$i + 1}) {
+      display: none;
+    }
+  }
+}
+
+// Styles
+.calendar {
+  margin: auto;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $border-size;
+  padding: $padding-size;
+  max-width: $calendar-width;
+
+  &__day {
+    width: $cell-size;
+    height: $cell-size;
+    border: $border-size solid $black;
+    background-color: $gray;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: $font-size;
+    text-align: center;
+    position: relative;
+
+    &::before {
+      content: "";
+      font-size: 30px;
+      font-family: Arial, Helvetica, sans-serif;
+      position: absolute;
+      top: 50%;
+      left: 50%;
```

Comment:
1. you don't need to use position absolute here, try to position it using display flex for `&__day`
2. it's a good practice to set font-family for the whole body